src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/RuntimePlatformCheckAnalyzerTests.cs

```diff
@@ -0,0 +1,785 @@
+        {
+            {|#2:M2()|};        // Platform checks:'IsOSPlatformEarlierThan;Linux;1.1 && IsOSPlatformOrLater;Windows;1'
+        }
+        else
+        {
+            {|#3:M2()|};        // Platform checks:'!IsOSPlatformEarlierThan;Linux;1.1 && IsOSPlatformOrLater;Windows;1'
+        }
+
+        {|#4:M2()|};        // Platform checks:''
+    }
+
+    void M2()
+    {
+    }
+}" + PlatformCheckApiSource;
+
+            await VerifyCS.VerifyAnalyzerAsync(source,
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(0).WithArguments(""),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(1).WithArguments("!IsOSPlatformOrLater;Windows;1"),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(2).WithArguments("IsOSPlatformEarlierThan;Linux;1.1 && IsOSPlatformOrLater;Windows;1"),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(3).WithArguments("!IsOSPlatformEarlierThan;Linux;1.1 && IsOSPlatformOrLater;Windows;1"),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(4).WithArguments(""));
+        }
+
+        [Fact]
+        public async Task SimpleIfTestWithLogicalAnd()
+        {
+            var source = @"
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+class Test
+{
+    void M1()
+    {
+        {|#0:M2()|};    // Platform checks:''
+
+        if(RuntimeInformationHelper.IsOSPlatformOrLater(OSPlatform.Windows, 1) &&
+           RuntimeInformationHelper.IsOSPlatformEarlierThan(OSPlatform.Windows, 2))
+        {
+            {|#1:M2()|};        // Platform checks:'IsOSPlatformEarlierThan;Windows;2 && IsOSPlatformOrLater;Windows;1'
+        }
+
+        {|#2:M2()|};        // Platform checks:''
+    }
+
+    void M2()
+    {
+    }
+}" + PlatformCheckApiSource;
+
+            await VerifyCS.VerifyAnalyzerAsync(source,
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(0).WithArguments(""),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(1).WithArguments("IsOSPlatformEarlierThan;Windows;2 && IsOSPlatformOrLater;Windows;1"),
+                VerifyCS.Diagnostic(RuntimePlatformCheckAnalyzer.Rule).WithLocation(2).WithArguments(""));
+        }
+
+        [Fact]
+        public async Task SimpleIfElseTestWithLogicalAnd()
+        {
+            var source = @"
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+class Test
+{
+    void M1()
+    {
+        {|#0:M2()|};    // Platform checks:''
+
+        if(RuntimeInformationHelper.IsOSPlatformOrLater(OSPlatform.Windows, 1) &&
+           RuntimeInformationHelper.IsOSPlatformOrLater(OSPlatform.Linux, 1))
+        {
+            {|#1:M2()|};        // Platform checks:'IsOSPlatformOrLater;Linux;1 && IsOSPlatformOrLater;Windows;1'
+        }
+        else
+        {
+            {|#2:M2()|};        // Platform checks:''
```

Comment:
This seems like a bug in the DFA - this should detect negation of checks above. I'll try to fix it in follow-up.

Reply:
What you think it should to detect by negation of checks above? I can't say because there AND logic used, for me this evaluation is righ

Reply:
I was thinking it should have `!IsOSPlatformOrLater;Linux;1 || !IsOSPlatformOrLater;Windows;1`, but you are right, that is not any valuable data.